nixos/modules/config/system-environment.nix

```diff
@@ -23,7 +23,7 @@ in
         strings.  The latter is concatenated, interspersed with colon
         characters.
       '';
-      type = types.attrsOf (types.loeOf types.str);
+      type = with types; attrsOf (either str (listOf str));
```

Comment:
That is puzzling, if the two variables are list of strings, they should be merged by the either merge function.

I tried with the following configuration.nix:

```nix
{ config, pkgs, ... }:
{
  hardware.opengl.enable = true;
  hardware.sane.enable = true;
}
```

And it generate what seems to be a correct `pam-environment`:

```
LANG="en_US.UTF-8"
LD_LIBRARY_PATH="/nix/store/6vc7djgg34kp3lqlgb9nadk5kjq5yimh-sane-config/lib/sane:/run/opengl-driver/lib:/run/opengl-driver-32/lib"
LOCALE_ARCHIVE="/run/current-system/sw/lib/locale/locale-archive"
NIX_CONF_DIR="/etc/nix"
NIX_OTHER_STORES="/run/nix/remote-stores/*/nix"
NIX_PATH="nixpkgs=/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs:nixos-config=/etc/nixos/configuration.nix:/nix/var/nix/profiles/per-user/root/channels"
SANE_CONFIG_DIR="/nix/store/6vc7djgg34kp3lqlgb9nadk5kjq5yimh-sane-config/etc/sane.d"
TZDIR="/etc/zoneinfo"
```

Could you provide a configuration that trigger this problem?